### PR TITLE
[workload]Avoid unnecessary allocation in HasAllChecks

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Avoids unnecessary allocation in workload.HasAllChecks
#### Which issue(s) this PR fixes:

Relates to: #1090

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/workload/workload.go [hunk 1]
```diff
@@ -449,6 +449,10 @@ func HasAllChecks(wl *kueue.Workload, mustHaveChecks sets.Set[string]) bool {
 		return true
 	}
 
+	if len(mustHaveChecks) > len(wl.Status.AdmissionChecks) {
+		return false
+	}
```
Comment (tenzen-y) [inline pkg/workload/workload.go:454]: Why do we use `len(mustHaveChecks)` here? We use `mustHaveChecks.Len()` in other places such as L448, L460.

**Review by tenzen-y — COMMENTED** [review]
Just confirmation.

/approve

File: pkg/workload/workload.go (hunk 1 above)
Comment (trasc) [reply pkg/workload/workload.go]: I forgot it's a set :). Done.

**Pull request merged**